pages/a/monitor-your-cratedb-cluster.html

```diff
@@ -0,0 +1,107 @@
+
+## Importance of Metrics
+
+Metrics for your database should provide a high level overview on what is going on in the database. To understand what metrics should be monitored in your CrateDB, some in-depth knowledge about CrateDB's internal workings is required. There are two important scenarios that require CrateDB specific knowledge: Insert/update/delete and queries. Fundamentally, CrateDB uses [Lucene][1] indices underneath. Simplified, an index is a file on disk that stores an [inverted index](https://en.wikipedia.org/wiki/Inverted_index) as an immutable array to binary search on. Consequently there are no real updates or deletes - only inserts and merges. Inserts add new segments which are eventually merged (can be executed manually by running [OPTIMIZE TABLE](/docs/reference/sql/reference/optimize.html) or will be triggered as needed by Lucene), where documents that are marked for deletion are skipped and the old segments are deleted through this process. Practically, this means that update processes are managed through inserts, with documents to be deleted being left out of the update process. These processes are the primary cause of disk I/O. Additionally, deleting or updating with a condition attached will use memory and may result in into heap memory issues, since they also work just like queries. The actual amount of heap space required is determined by the nature of the query itself, since for aggregations the homogeneity of a field's values is important, whereas for projection queries the sizes of the selected fields matters most. Additionally 'handler nodes' ultimately apply Limit and Offset (and sometimes merge results), causing the node to require more memory than other participants.
+
+## Performance Monitoring
+
+The most important part of monitoring a CrateDB cluster is the understanding of which metrics you care about and why. At Crate.io we are trying to keep things simple and we have included some fundamental metrics in the AdminUI.
+
+![CrateDB monitors CPU, heap, and disk](/static/images/1116/crate-db-load.png)
+
+The numbers provided indicate a general load on the machine as well as within CrateDB:
+- CPU (overall and CrateDB) usage
+- Heap usage
+- Disk usage
+- Disk I/O
+
+For use with CrateDB we recommend monitoring at least the numbers listed above in order to better understand the health of cluster. It can make sense for large clusters to monitor a node's network connection to see if the infrastructure is causing a bottleneck or to identify nodes with an unusual amount of client connections. For basic monitoring purposes, standard Linux tools are a great way to get the number of active connections: `netstat -an | grep :4200 | wc -l` provides the number of active TCP connections on port 4200, ([CrateDB's HTTP port](https://crate.io/docs/reference/configuration.html#ports)).
```

Comment:
I would also mention that most of the common monitoring tools like scout, nagios,... support those things.